src/main/java/io/github/syst3ms/skriptparser/effects/EffPrint.java

```diff
@@ -34,7 +35,7 @@ public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContex
 
     @Override
     public void execute(TriggerContext ctx) {
-        String[] strs = string.getValues(ctx);
+        String[] strs = TagManager.occasionally(() -> string.getValues(ctx));
```

Comment:
I still do not understand what this "occasional" thing is about. I hope it's useful, too, because I frown upon global state like this.